README.md

```diff
@@ -1,2 +1,69 @@
-# viper-ios-demo
-This is a demo app using VIPER architecture.
+# VIPER Architecture
+
+VIPER is an application of **Clean Architecture** to iOS apps. 
+
+The word VIPER is a backronym for `View`, `Interactor`, `Presenter`, `Entity`, and `Router`
+
+With VIPER applied, the code is easy to understand, easier to test, and as a result, also easier to maintain.
+
+## Main Parts
+
+- **Entity**: contains basic model objects used by the Interactor.
+- **Interactor**: contains the *business logic* as specified by a use case
+- **Presenter**: contains view logic for *preparing content for display* (as received from the Interactor) and for *reacting to user inputs* (by requesting new data from the Interactor).
+- **Router**: contains *navigation logic* for describing which screens are shown in which order.
+- **View**: displays what it is told to by the Presenter and relays user input back to the Presenter.
+
+![viper-parts](./resources/images/viper-parts.jpg)
+
+## How We Work With VIPER?
+
+For each module, we have a `Module` class, that will connect all parts of VIPER architecture.
+
+![module](./resources/images/module-parts.jpg)
+
+### In a Module
+
+![each-module](./resources/images/each-module.jpg)
+
+### Module connect to another Module or Submodule
+
+![each-module](./resources/images/modules.jpg)
```

Comment:
Main module is holding submodule's input (presenter) with strong reference. 

Submodule's output ( kind of submodule's delegate ) should hold output with weak reference ✌️

![picture](https://i.imgur.com/hiK9wMJ.png)